static/js/settings_emoji.js

```diff
@@ -52,13 +52,32 @@ exports.reset = function () {
     meta.loaded = false;
 };
 
+exports.sort_but_user_emoji_on_top = function () {
+    const emoji_data = Object.values(page_params.realm_emoji);
+    emoji_data.sort(function (a, b) {
+        if (a.name > b.name) {
+            return 1;
+        } else if (a.name === b.name) {
+            return 0;
+        }
+        return -1;
+    });
+    for (let index = 0; index < emoji_data.length; index = index + 1) {
+        if (emoji_data[index].author.full_name === page_params.full_name) {
+            emoji_data.unshift(emoji_data[index]);
+            emoji_data.splice(index + 1, 1);
+        }
+    }
```

Comment:
This is an awkward way to sort with `unshift/splice`; why not just encode the preference for emoji the user uploaded in the sorting function?